src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,115 @@
+keySetMeta elektra/spec/impl fallback a (Key P.Nothing) = a \n fallback _ a = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/override/#\
+keySetMeta elektra/spec/type RegexContains b a => Key b :: . -> Key a -> Key a \
+keySetMeta elektra/spec/impl override (Key P.Nothing) b = b \n override a _ = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/check/range \
+keySetMeta elektra/spec/type RegexIntersects a b => P.Proxy b :: Range . -> Key a -> Key (RegexIntersection a b) \
+keySetMeta elektra/spec/impl checkrange _ a = a \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/elektra/spec/check/long \
+keySetMeta elektra/spec/type RegexIntersects a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]" => Key a -> Key (RegexIntersection a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]") \
+keySetMeta elektra/spec/impl checklong a = a \
+ksAppendKey\
+keyClear \
+keySetName spec/examples/simplespecification/key1 \
+keySetMeta check/range 0-5000 \
+ksAppendKey \
+keyClear \
+keySetName spec/examples/simplespecification/key2 \
+keySetMeta check/range 7200-10000 \
+ksAppendKey\
+keyClear \
+keySetName spec/examples/simplespecification/key3 \
+keySetMeta check/long \
+keySetMeta fallback/#1 spec/examples/simplespecification/key1 \
+ksAppendKey \
+keyClear \
+kdbSet spec/examples/simplespecification' | kdb shell
+
+kdb get spec/examples/simplespecification/key1
+```
+
+Add an invalid link and see how it refuses the specification, showing the erroneous
+parts instead. Note that currently it doesn't get active when using setmeta though.
+
+```sh
+kdb setmeta spec/examples/simplespecification/key2 fallback/#1 spec/examples/simplespecification/key1
+
+kdb get spec/examples/simplespecification/key2
+# STDERR-REGEX: .*Couldn't match type.*
```

Comment:
error on getting? It should yield an error setting.

Reply:
fixed